client/rest/src/main/java/org/elasticsearch/client/RestClient.java

```diff
@@ -840,6 +965,11 @@ Response get() throws IOException {
                     e.initCause(exception);
                     throw e;
                 }
+                if (exception instanceof ConnectException) {
+                    ConnectException e = new ConnectException(exception.getMessage());
+                    e.initCause(exception);
+                    throw e;
```

Comment:
nit: I wonder if this should be a separate change, or maybe it is needed as part of this PR?

Reply:
It has been so long since I wrote it that I'm not sure where this one came from. I imagine I hit it while testing something months ago. I can factor it out if you'd like.